### PR TITLE
Assign system-cluster-critical priority to operator and component

### PR DESCRIPTION
**Pull request by ironcladlou**
Apply the `system-cluster-critical` pod priority class to the operator and
CoreDNS deployments in accordance with the designation of cluster DNS as
a core feature.

Fixes https://github.com/openshift/cluster-dns-operator/issues/57

### CODE REVIEW

**Comment by ironcladlou** [comment]
@openshift/sig-network-edge PTAL

File: assets/dns/daemonset.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ spec:
         openshift-app: dns
     spec:
       serviceAccountName: dns
+      priorityClassName: system-cluster-critical
```
Comment (Miciah) [inline assets/dns/daemonset.yaml]: Should this be `system-node-critical`? Without the operator, the cluster is broken; without the local CoreDNS instance, the node is broken.

Comment (ironcladlou) [reply assets/dns/daemonset.yaml]: Hmm, could be, especially now that we're responsible for image registry connectivity via /etc/hosts on the node. @aveshagarwal, thoughts?

Comment (pravisankar) [reply assets/dns/daemonset.yaml]: Yes, I think this should be `system-node-critical` (should never be evicted from the node).

Comment (aveshagarwal) [reply assets/dns/daemonset.yaml]: +1 for system-node-critical for dns pods.

Comment (ironcladlou) [reply assets/dns/daemonset.yaml]: Updated to use system-node-critical

**Comment by Miciah** [comment]
/lgtm
Besides flakes, it looks like e2e-aws is blocked on https://github.com/openshift/cluster-dns-operator/pull/58.

**Pull request merged**